src/DateRangePicker/internal/Day/Day.tsx

```diff
@@ -0,0 +1,56 @@
+import { Dayjs } from "dayjs";
+import { FC, memo, ReactNode } from "react";
+import { DayContainer } from "./styled";
+
+type DateRange = {
+  startDate: Dayjs | null;
+  endDate: Dayjs | null;
+};
+
+type Props = {
+  selected: boolean;
+  value: Dayjs;
+  date: DateRange;
+  clickState: "start" | "end" | "none";
+  changeState: (newState: "start" | "end" | "none") => void;
+  onClickDate?: (newDate: DateRange) => void;
+  children: ReactNode;
+};
+
+export const Day: FC<Props> = memo(
+  ({
+    selected,
+    date,
+    value,
+    clickState,
+    changeState,
+    onClickDate,
+    children,
+  }) => {
+    return (
+      <DayContainer
+        selected={selected}
+        // How can I improve the User Experience?
+        // Answer: impossible
```

Comment:
Impossible to improve UX (._.)

Reply:
Now my local code looks like this, but I can't say that the experience is good.
The state that the program expects and the state that the user is trying to enter do not always match, and I think it will never be possible for the user to reconcile the range selection.

```tsx
export const Day: FC<Props> = memo(
  ({
    selected,
    isBetween,
    date,
    value,
    clickState,
    changeState,
    onClickDate,
    children,
  }) => (
    <DayContainer
      selected={selected}
      isBetween={isBetween}
      // How can I improve the User Experience?
      // Answer: impossible
      onClick={() => {
        if (clickState === "start") {
          if (value.isAfter(date.endDate)) {
            onClickDate?.({
              startDate: date.endDate,
              endDate: value,
            });
          } else {
            onClickDate?.({
              startDate: value,
              endDate: date.endDate,
            });
          }
          changeState("end");
        } else if (clickState === "end") {
          if (value.isBefore(date.startDate)) {
            onClickDate?.({ startDate: value, endDate: date.startDate });
          } else {
            onClickDate?.({ startDate: date.startDate, endDate: value });
          }
          changeState("start");
        }
      }}
    >
      {children}
    </DayContainer>
  )
);

```